effects/push-cache/default.nix

```diff
@@ -0,0 +1,160 @@
+        default = false;
+        description = ''
+          Enables an effect that pushes certain outputs to a different binary cache.
+
+          Hercules CI normally pushes everything to the cache(s) configured on the agent. This effect supplements that behavior by letting you push a subset of those to a different cache.
+          Note that it only pushes the output closure, and not the closures of build dependencies used during the build stage of the CI job. (Unless those closures happen to also be part of the output or "runtime" closure)
+        '';
+      };
+      attic-client-pkg = lib.mkOption {
+        type = lib.types.package;
+        description = ''
+          Version of the attic-client package to use on \"x86_64-linux\".
+
+          Hint: You can use `attic.packages.x86_64-linux.attic-client` from the attic flake.
+        '';
+        default = pkgs-x86_64-linux.attic-client or (throw
+          "push-cache-effect.attic-client-pkg: It seems that attic hasn't been packaged in Nixpkgs (yet?). Please check <nixpkgs packaging request issue> or set <option> manually.");
+      };
+      cachix-pkg = lib.mkOption {
+        type = lib.types.package;
+        default = pkgs-x86_64-linux.cachix;
+        description =
+          ''Version of the cachix package to use on "x86_64-linux".'';
+      };
+      caches = lib.mkOption {
+        description = ''
+          An attribute set, each `name: value` pair translates to
+          an effect under `onPush.default.outputs.effects.push-cache-effect.name`.
+        '';
+        example = ''
+          {
+            our-cachix = {
+              type = \"cachix\";
+              secretName = \"our-cachix-token\";
+              branches = [ \"master\" ];
+              packages = [ pkgs.hello ];
+            };
+          }
+        '';
+        type = lib.types.attrsOf (lib.types.submodule ({ name, ... }: {
+          options = {
+            name = lib.mkOption {
+              type = lib.types.str;
+              default = name;
+              description = ''
+                Name of the effect. By default it's the attribute name.
+              '';
+            };
+            type = lib.mkOption {
+              type = lib.types.enum [ "attic" "cachix" ];
+              description = ''A string "attic" or "cachix".'';
+            };
+            packages = lib.mkOption {
+              type = with lib.types; listOf package;
+              description = "List of packages to push to the cache.";
+              example = "[ pkgs.hello ]";
+            };
+            secretName = lib.mkOption {
+              type = lib.types.str;
+              description = ''
+                Name of the HerculesCI secret. See [HerculesCI docs](https://docs.hercules-ci.com/hercules-ci-agent/secrets-json).
+                The secrets "data" field should contain given data:
+
+                ```
+                  "data": {
+                    "name": "my-cache-name",
```

Comment:
Should `name` and `endpoint` be in the secret? I think the attribute name can be used as `name`, and the `endpoint` doesn't need to be secret, it can be an option. But it's convenient to manage together with the `token`.

Reply:
I think both are acceptable solutions.